mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/SearchThreadIdGuessingAlgorithm.java

```diff
@@ -67,6 +67,18 @@ public SearchThreadIdGuessingAlgorithm(MailboxManager mailboxManager, MessageIdM
             .switchIfEmpty(Mono.just(ThreadId.fromBaseMessageId(messageId)));
     }
 
+    @Override
+    public Flux<MessageId> getMessageIdsInThread(ThreadId threadId, MailboxSession session) throws MailboxException {
+        MultimailboxesSearchQuery expression = MultimailboxesSearchQuery.from(SearchQuery.matchAll()).build();
+
+        return Flux.from(mailboxManager.search(expression, session, Long.MAX_VALUE))
+            .collectList()
+            .flatMapMany(messageIds -> messageIdManager.getMessagesReactive(messageIds, FetchGroup.MINIMAL, session))
+            .filter(messageResult -> messageResult.getThreadId().equals(threadId))
```

Comment:
TODO (Another PR)

Push this to the search engine.

How: 

`SearchQuery::withThreadId` and implement it for scanning search, lucene search as well as ElasticSearch.